firedrake/preconditioners/pytorch_coupling/pytorch_custom_operator.py

```diff
@@ -0,0 +1,74 @@
+import collections
+from functools import partial
+
+from firedrake.preconditioners.pytorch_coupling import get_backend
+from firedrake.function import Function
+
+from pyadjoint.reduced_functional import ReducedFunctional
+
+
+backend = get_backend('pytorch')
```

Comment:
if backend:
    BackendFunction = ...
else:
    dummy class that exceptions on instantiation.